### PR TITLE
Added packaging to the installation.rst requirements.

### PR DESCRIPTION
**Pull request by AGaliciaMartinez**
**Changelog**
Added packaging to the requirements in `installation.rst`.
Update requirements.txt with the information in setup.cfg.

### CODE REVIEW

**Comment by AGaliciaMartinez** [comment]
I realised it was not necessary for me to include setuptools and wheel in the installation. I wonder if this is because setuptools is being taken from the system installation instead of the virtualenv? 

Also, I noticed that requirements.txt does not include stuptools nor wheel. Should I include them?

**Comment by jakelishman** [comment]
`requirements.txt` has no special meaning at all so it doesn't really matter, but typically it's used for runtime, not build time, so it doesn't need `setuptools` or `wheel`.  `packaging` is required at both, but I had actually thought it was a dependency of `setuptools`.  Still, we use it explicitly, so best add it.

The actual main source-of-truth runtime requirements are in `setup.cfg`, and the build requirements are in `pyproject.toml`.

**Comment by AGaliciaMartinez** [comment]
Thank you Jake for taking a quick look!
I checked once more and the following produces an error (used in an "empty" virtual environment):
```
> pip install setuptools wheel 'cython>=0.29.20' 'numpy>=1.16.6,<1.20' 'scipy>=1.0'
> python setup.py develop

Traceback (most recent call last):
  File "/home/kaladin/Documents/git/qutip/qutip4/setup.py", line 13, in <module>
    import packaging.version
ModuleNotFoundError: No module named 'packaging'
```
So I think it is not included with `setuptools`. 

> `requirements.txt` has no special meaning at all so it doesn't really matter,
Lets not add wheels and setuptools to the requirements then. Updating it with the latest dependency versions in setup.cfg dependencies should be fine though.

**Comment by AGaliciaMartinez** [comment]
The test for python 3.6 seems to fail at coveralls, not sure why though. It may not be related as the rest of the tests do not fail. Could it be rerun?

**Comment by jakelishman** [comment]
Sounds good to me.  Yeah, even if `packaging` _were_ in `setuptools`, it'd still be an explicit dependency of ours, so you're right to put it in, thanks.

Yeah, that coveralls thing just looked like a network timeout on their side.  Not the fault of this PR.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cython>=0.21
-numpy>=1.12
+cython>=0.29.20
+numpy>=1.16.6,<1.20
```
Comment (jakelishman) [inline requirements.txt:2]: There's no upper limit on Numpy at runtime, unless something's changed recently.

Comment (AGaliciaMartinez) [reply requirements.txt]: Indeed the runtime does not have upper limit, only the building requirements do. Ive just updated `requirements.txt`.

**Review by hodgestar — APPROVED** [review]

**Pull request merged**